homeassistant/components/rointe/climate.py

```diff
@@ -0,0 +1,227 @@
+"""Support for Rointe Climate."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from rointesdk.device import RointeDevice
+
+from homeassistant.components.climate import (
+    PRESET_COMFORT,
+    PRESET_ECO,
+    ClimateEntity,
+    ClimateEntityDescription,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, LOGGER, RointeCommand, RointeOperationMode, RointePreset
+from .coordinator import RointeDataUpdateCoordinator
+from .entity import RointeRadiatorEntity
+
+AVAILABLE_HVAC_MODES: list[HVACMode] = [HVACMode.OFF, HVACMode.HEAT, HVACMode.AUTO]
+AVAILABLE_PRESETS: list[str] = [
+    RointePreset.ECO,
+    RointePreset.COMFORT,
+    RointePreset.ICE,
+]
+
+ROINTE_HASS_MAP = {
+    RointePreset.ECO: PRESET_ECO,
+    RointePreset.COMFORT: PRESET_COMFORT,
+    RointePreset.ICE: RointePreset.ICE,
+}
+
+RADIATOR_TEMP_STEP = 0.5
+RADIATOR_TEMP_MIN = 7.0
+RADIATOR_TEMP_MAX = 30.0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the radiator climate entity from the config entry."""
+    coordinator: RointeDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    # Register the Entity classes and platform on the coordinator.
+    coordinator.add_entities_for_seen_keys(
+        async_add_entities, [RointeHaClimate], "climate"
+    )
+
+
+class RointeHaClimate(RointeRadiatorEntity, ClimateEntity):
+    """Climate entity."""
+
+    _attr_icon = "mdi:radiator"
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_target_temperature_step = RADIATOR_TEMP_STEP
+    _attr_hvac_modes = AVAILABLE_HVAC_MODES
+    _attr_preset_modes = AVAILABLE_PRESETS
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        radiator: RointeDevice,
+        coordinator: RointeDataUpdateCoordinator,
+    ) -> None:
+        """Init the Climate entity."""
+
+        super().__init__(coordinator, radiator, unique_id=radiator.id)
+
+        self.entity_description = ClimateEntityDescription(
+            key="radiator",
+            name=radiator.name,
+        )
```

Comment:
This isn't needed. I think you added this to be able to set the name. Instead you can create a DeviceInfo with the radiator name as name, and because you have `_attr_name = None`, the name of the entity will be set to the name of the device.

Reply:
The `DeviceInfo` is provided by `RointeRadiatorEntity`.

`class RointeHaClimate(RointeRadiatorEntity, ClimateEntity):`